### PR TITLE
mgr/cephadm: prefix daemon ids with hostname

### PR DESCRIPTION
**Pull request by liewegas**
We have two options here:

1. ``mds.fsname.host-abcdef``.  This is the current behavior.  It makes the "id" $host-$random, and uses - as a separator so that this does not pick up config settings in the ``mds.fsname.host`` section.  (Current behavior.)
2. ``mds.fsname.host.abcdef``.  This would let you set per-host options for daemons in this group.  I think it blurs things a bit, but I can also see that it could be a useful thing.

??

### CODE REVIEW

**Comment by jdurgin** [comment]
option 2 does seem useful - you could set some config that would apply to the host before adding services to it

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -743,12 +743,14 @@ def get_unique_name(self, existing, prefix=None, forcename=None):
                 raise RuntimeError('specified name %s already in use', forcename)
             return forcename
 
+        if '.' in host:
+            host = host.split('.')[0]
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/module.py]: maybe add a warning that hosts not should not be FQDNs?

**Comment by sebastian-philipp** [comment]
`mds.fsname.host.abcdef` lgtm

**Comment by liewegas** [comment]
https://github.com/ceph/ceph/pull/33042

**Pull request merged**